llvm/lib/Passes/PassBuilderPipelines.cpp

```diff
@@ -1076,6 +1078,11 @@ PassBuilder::buildModuleSimplificationPipeline(OptimizationLevel Level,
 
   ModulePassManager MPM;
 
+  if (Phase != ThinOrFullLTOPhase::ThinLTOPostLink) {
+    MPM.addPass(NameAnonGlobalPass());
```

Comment:
Currently this is invoked at the very end of the pre-LTO pipelines (just before we will build the module summary) via addRequiredLTOPreLinkPasses. If it is going to be done early then it can presumably be removed from there. Is it possible for an anon global to be introduced in between this point and the end of the pre-LTO pipeline where we build the summary?